common/src/main/java/com/microsoft/identity/common/internal/platform/AndroidPlatformUtil.java

```diff
@@ -261,6 +263,31 @@ public static ArrayList<Map.Entry<String, String>> updateWithOrDeleteWebAuthnPar
         return result;
     }
 
+    /**
+     * Check if the host app is running within work profile.
+     * @param appContext current application context.
+     * @return true if app is in work profile, false if in personal profile or OS is below LOLLIPOP.
+     */
+    public static boolean isInWorkProfile(@NonNull final Context appContext) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
```

Comment:
Can we add comments explaining why we have different behavior based on android version

Reply:
Yes: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2561/commits/5d9292b6ff85353ec672098d89e81da16ff589bf